test/jdk/java/net/httpclient/HttpClientBuilderTest.java

```diff
@@ -267,24 +264,16 @@ static void testPriority() throws Exception {
     }
 
     /**
-     * Tests the {@link HttpClient,java.net.http.HttpClient.Builder#localAddress(SocketAddress)} method
+     * Tests the {@link HttpClient,java.net.http.HttpClient.Builder#localAddress(InetAddress)} method
      */
     @Test
     public void testLocalAddress() throws Exception {
         HttpClient.Builder builder = HttpClient.newBuilder();
         // setting null should work fine
         builder.localAddress(null);
-        // setting to InetSocketAddress with port 0 should work fine
-        builder.localAddress(new InetSocketAddress(0));
-        builder.localAddress(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
-        builder.localAddress(new JustAnotherInetSocketAddress(0));
+        builder.localAddress(InetAddress.getLoopbackAddress());
```

Comment:
We probably also need a MockHttpClientBuilder to test the behaviour of the default implementation (check that it throws UOE).

Reply:
I've now updated this PR to add a new test method which tests the default method implementation of `localAddress`.